activesupport/test/dependencies_test.rb

```diff
@@ -240,6 +240,46 @@ class AnotherConstant::ReloadError
     end
   end
 
+  # Regression see https://github.com/rails/rails/issues/32359
+  def test_explicit_unloadable_constants_can_be_unloaded_and_reloaded
+    with_loading "autoloading_fixtures/explicit_unloadable" do
+      # autoload this constant:
+      assert UnloadableExample
+
+      # Assert that AS::D's internal state matches Ruby's state:
+      assert defined?(UnloadableExample)
+      assert_equal ["UnloadableExample"], ActiveSupport::Dependencies.autoloaded_constants
+      assert_equal 1, ActiveSupport::Dependencies.explicitly_unloadable_constants.size
+      assert_equal 1, ActiveSupport::Dependencies.loaded.size
+      # Now cleanly unload the constant
+      ActiveSupport::Dependencies.remove_unloadable_constants!
+
+      # Since the constant was properly unloaded, it can be reloaded:
+      assert UnloadableExample
+    end
+  end
+
+  # Regression see https://github.com/rails/rails/issues/32359
+  def test_explicit_unloadable_constants_can_are_cleanly_unloaded
+    with_loading "autoloading_fixtures/explicit_unloadable" do
+      # autoload this constant:
+      assert UnloadableExample
+
+      # Assert that AS::D's internal state matches Ruby's state:
+      assert defined?(UnloadableExample)
+      assert_equal ["UnloadableExample"], ActiveSupport::Dependencies.explicitly_unloadable_constants
```

Comment:
Is there a reason you match against the array here, but only check the size on 252?

Reply:
No good  reason -- I was just using the test suite as I worked, and a length mismatch wasn't enough information to realize why my test case wasn't failing properly! So I spelled out the whole array here for a more helpful error message when the assertion failed. Happy to standardize these if it's important to do so!